text/0059-nodes-capabilities-discovery.md

```diff
@@ -0,0 +1,142 @@
+# RFC-0059: Add a discovery mechanism for nodes based on their capabilities
+
+|                 |                                                                                             |
+| --------------- | ------------------------------------------------------------------------------------------- |
+| **Start Date**  | 2023-12-18                                                                                  |
+| **Description** | Nodes having certain capabilities register themselves in the DHT to be discoverable         |
+| **Authors**     | Pierre Krieger                                                                              |
+
+## Summary
+
+This RFC proposes to make the mechanism of [RFC #8](https://github.com/polkadot-fellows/RFCs/blob/main/text/0008-parachain-bootnodes-dht.md) more generic by introducing the concept of "capabilities".
+
+Implementations can implement certain "capabilities", such as serving old block headers or being a parachain bootnode.
+
+The discovery mechanism of RFC #8 is extended to be able to discover nodes of specific capabilities.
+
+## Motivation
+
+The Polkadot peer-to-peer network is made of nodes. Not all these nodes are equal. Some nodes store only the headers of recently blocks, some nodes store all the block headers and bodies since the genesis, some nodes store the storage of all blocks since the genesis, and so on.
```

Comment:
```suggestion
The Polkadot peer-to-peer network is made of nodes. Not all these nodes are equal. Some nodes store only the headers of recent blocks, some nodes store all the block headers and bodies since the genesis, some nodes store the storage of all blocks since the genesis, and so on.
```